controller/src/beerocks/master/db/db.cpp

```diff
@@ -3044,6 +3044,25 @@ void db::lock() { db_mutex.lock(); }
 
 void db::unlock() { db_mutex.unlock(); }
 
+//
+// certification
+//
+
+void db::setting_certification_mode(bool en)
+{
+    if (!en) {
+        remove_certification_tx_buffer();
```

Comment:
Why is all this complicated certification_tx_buffer handling needed? Why don't we just follow the same pattern as e.g. dfs_reentry?

As far as I can see, it is not really used anywhere. It is filled from the CMDU_TX but otherwise ignored, it seems.

Whenever something is not immediately obvious, it should be explained either in a comment or (usualy) in the commit message. The commit message that adds the certification_tx_buffer doesn't even mention this, it just talks about the flag and the DB API change.

Reply:
We could choose to declare a static buffer or allocate and deallocate it according to the certification flag. I asked Tomer and he preferred to allocate and deallocate it.
I don't understand why it is not obvious to you.